src/safeposix/syscalls/net_calls.rs

```diff
@@ -1586,30 +1661,44 @@ impl Cage {
                                         panic!("Unknown errno value from socket listen returned!")
                                     }
                                 };
+                                //Remove the tuple of the address, port, and 
+                                //port type from the set of listening ports 
+                                //as we are returning from an error
+                                //** Why dont we use 'porttuple' as the argument for the key ?? */
                                 NET_METADATA.listening_port_set.remove(&mux_port(
                                     ladr.addr().clone(),
                                     ladr.port(),
                                     sockhandle.domain,
                                     TCPPORT,
                                 ));
+                                //Set the socket state to NOTCONNECTED, as
+                                //the socket is not listening
                                 sockhandle.state = ConnState::NOTCONNECTED;
                                 return lr;
                             };
 
-                            //set rawfd for select
+                            //set rawfd for select sys call
+                            //** Why is this being done in listen ?? 
```

Comment:
I guess since there is a code logic before that does something like "if the socket didn't bind to any address, then assign a new address". So the innersocket could be possibly a new one that just created in this syscall, and we certainly want to update its rawfd as well